src/components/organisms/MapBoxMap/MapboxMap.tsx

```diff
@@ -44,57 +55,175 @@ const MapboxMap: React.FC = () => {
+        refetchOnWindowFocus: false,
+    })
+
     useEffect(() => {
         const map = getGlobeMap(mapContainerRef)
         setMap(map)
         loadStyles(map)
         addNavigationControls(map)
 
-        fetchRegions()
-            .then((regions) => {
-                addRegionLayer(map, regions)
-                map.on('click', 'regions', (e) => {
-                    setRegionId(e.features![0].properties!.id)
-                    const regionName = e.features![0].properties!.name
-                    const regionSize = e.features![0].properties!.area_km2
-
-                    const aoiPolygon: Polygon = JSON.parse(e.features![0].properties!.polygon)
-
-                    fetchRegionDatetimes(regionId).then((regionDatetimes) => {
-                        map.setLayoutProperty('regions', 'visibility', 'none')
-                        map.fitBounds(getBoundingBox(aoiPolygon))
-
-                        addPolygonLayer(map, aoiPolygon)
-                        openSidebar()
-                        const jobs = regionDatetimes.map((regionDatetimes) => regionDatetimes.timestamp)
-                        const regionProps: IRegionProperties = {
-                            id: regionId,
-                            name: regionName,
-                            jobs: jobs,
-                            areaSize: regionSize,
-                        }
-                        setRegionProps(regionProps)
-                        addPredictionLayer(map, jobs[0], regionId)
-                    })
-                })
-            })
-            .catch((error) => console.error('Failed to load regions on map:', error))
-
         return () => map.remove()
     }, [])
 
+    const { isPending: fetchingAoiCenters, data } = useQuery({
+        queryKey: ['regions'],
+        // queryFn: fetchRegions,
+        queryFn: async () => await fetchRegions(),
+    })
+
+    useEffect(() => {
+        if (data && map) {
+            loadRegionsOnMap(data)
+        }
+    }, [data, map])
+
+    //fails on hot reload, page reload works fine
+    function loadRegionsOnMap(regions: any) {
+        if (!map) {
+            console.error('Map is not initialized')
+            return
+        }
+        addRegionLayer(map, regions)
+
+        map.on('click', 'regions', (e) => {
+            const regionId = e.features![0].properties!.id
+            const regionName = e.features![0].properties!.name
+            const regionSize = e.features![0].properties!.area_km2
+            const regionPolygon: Polygon = JSON.parse(e.features![0].properties!.polygon)
+
+            const regionData: IRegionData = {
+                id: regionId,
+                name: regionName,
+                areaSize: regionSize,
+                polygon: regionPolygon,
+            }
+
+            //this causes the react query hook to refetch the data with the new regionId
+            setRegionId(e.features![0].properties!.id)
+            setRegionData(regionData)
```

Comment:
this is a bit hard to understand, without your comment it would be hard to realize that assigning a region id is triggering a requst. Why not put the react query into a function then you can call it wherever you need which is easier to understand.